src/lib.rs

```diff
@@ -344,6 +346,10 @@ impl<Types: IpfsTypes> std::ops::Deref for Ipfs<Types> {
 }
 
 impl<Types: IpfsTypes> Ipfs<Types> {
+    pub fn peer_id(&self) -> PeerId {
+        self.keys.get_ref().public().into_peer_id()
+    }
+
```

Comment:
Why add this method compared to sugaring the multiaddrs on `identity()` once the `Vec<Multiaddr>` arrives like already for the public key?

Reply:
It's just a helper used in `IpfsEvent::{GetAddresses, Listeners}` so as to not pass another message from `Ipfs` to `IpfsFuture`.

Reply:
It will be obviated once [this alternative](https://github.com/rs-ipfs/rust-ipfs/pull/110#discussion_r470702916) is implemented.